File: monitor-manage/internal/service/judgment/judgment.go
package judgment

import (
	"gitee.com/zekeGitee_admin/tx_gdut_monitor/monitor-manage/internal/model"
)

// IJudgment 服务接口化，对外提供Update方法，内部实际调用Grpc，尽可能解耦
type IJudgment interface {
	GetConfigsWithMetrics(ip string, local string, metrics []string) ([]model.JudgmentConfig, error)

	GetConfigs(ip string, local string) (map[string]model.JudgmentConfig, error)
	// Update 更新判定服务配置，一方面更新数据库，一方面告知判定服务rpc
	Update(ID int32, IP string,Local string, Metric string, Method int32, Period string, Threshold string) error
	// Del 删除指定id的配置
	Del(ip string, local string, id int32) error
}

File: monitor-web/monitor-txui/src/api/judgment.js
import { fetchGet, fetchPost } from '@/request.js'

// 获取指定agent的判定规则信息
export function GetAgentRule (ip, local) {
  let url = '/judgment/info/' + ip + '/' + local
  return fetchGet(url, {})
}

// 更新agent的指定id判定规则信息
export function UpdateRule (ip, local, id, method, metric, period, threshold) {
  let url = '/judgment/update/' + ip + '/' + local + '/' + id
  console.log(ip, local, id, method, metric, period, threshold)
  return fetchPost(url, {
    'method': method,
    'metric': metric,
    'period': period,
    'threshold': threshold
  })
}

// 删除agent的指定id判定规则信息
export function DelRule (ip, local, id) {
  let url = '/judgment/del/' + ip + '/' + local + '/' + id
  return fetchPost(url, {})
}
